src/System.Memory/src/System/Buffers/Text/Base64.cs

```diff
@@ -0,0 +1,179 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.Diagnostics;
+using System.Runtime.CompilerServices;
+using System.Runtime.Intrinsics;
+using System.Runtime.Intrinsics.X86;
+using Internal.Runtime.CompilerServices;
+
+namespace System.Buffers.Text
+{
+    // AVX2 version based on https://github.com/aklomp/base64/tree/e516d769a2a432c08404f1981e73b431566057be/lib/arch/avx2
+    // SSSE3 version based on https://github.com/aklomp/base64/tree/e516d769a2a432c08404f1981e73b431566057be/lib/arch/ssse3
+
+    public static partial class Base64
+    {
+        static Base64()
```

Comment:
It might be nice to get rid of the static constructor, if possible.

Reply:
It's possible, will do some benchmarks to see what impact this has.

Reply:
It is better to just initialize all the statics, all the time. The precise static constructor is a peanut-butter cost - it adds a bit of overhead to all static methods on the type.

Reply:
@jkotas sorry I can't follow. Keep the static constructor or not?

The reason why I've chosen the cctor is that in the methods there's a simple load from constant data into the simd-registers.
Initialization into static fields won't work, because it has to be taken into account wheter AVX2 / SSSE3 is available or not.

Reply:
Ah, nice and good to know. Thx!
(FYI: I'll continue tomorrow -- have to leave now).

Reply:
Done in ccf49eca3dc970c115fc6803bff1ab3c5203f534